VideoWeb/VideoWeb/ClientApp/src/app/waiting-space/waiting-room-shared/wr-participant-list-shared.component.ts

```diff
@@ -104,9 +105,50 @@ export abstract class WRParticipantStatusListDirective {
     }
 
     protected filterNonJudgeParticipants(): void {
-        this.nonJudgeParticipants = this.conference.participants.filter(
+        const nonJudgeParts = this.conference.participants.filter(
             x => x.role !== Role.Judge && x.role !== Role.JudicialOfficeHolder && x.hearing_role !== HearingRole.OBSERVER
         );
+
+        const individualWithInterpreterIndex = nonJudgeParts.findIndex(
+            x =>
+                x.role === Role.Individual &&
+                x.hearing_role !== HearingRole.INTERPRETER &&
+                Array.isArray(x.linked_participants) &&
+                x.linked_participants.length > 0
+        );
+        if (individualWithInterpreterIndex === -1) {
+            this.nonJudgeParticipants = nonJudgeParts;
+        } else {
+            this.nonJudgeParticipants = this.orderForInterpreter(nonJudgeParts, individualWithInterpreterIndex);
+        }
+    }
+
+    hasInterpreterLink(participant: ParticipantResponse) {
+        return participant.linked_participants.some(x => x.link_type === LinkType.Interpreter);
+    }
+
+    isInterpreter = function (participant: ParticipantResponse) {
+        return participant.hearing_role === HearingRole.INTERPRETER;
+    };
+
+    getInterpreteeName(interpreterId: string) {
+        const interpreter = this.nonJudgeParticipants.find(x => x.id === interpreterId);
+        return this.nonJudgeParticipants.find(x => x.id === interpreter.linked_participants[0].linked_id).name;
+    }
+
+    private orderForInterpreter(
```

Comment:
Perhaps something a bit more succinct like this

```ts
    private orderForInterpreter(
        participants: ParticipantResponse[],
        interpretee: number
    ): ParticipantResponse[] {
        const linkDetails = interpretee.linked_participants[0];
        const sortedParticipants = [interpretee];
        const interpreter = nonJudgeParticipants.find(x => x.id === linkDetails.linked_id);
        sortedParticipants.push(interpreter);
        return [...participants.filter(p => ![interpretee.id, interpreter.id].includes(p.id)), ...sortedParticipants];
    }
```

Reply:
not sure that code will work as interpretee is a number, so it won't contain linked_participants[0]

Reply:
but I get the gist

Reply:
sorry, I meant to make it a participant 👍

Reply:
done